### PR TITLE
[ci] Move duplicated "synthesize with Vivado" logic into a script

### PR DESCRIPTION
**Pull request by rswarbrick**

### CODE REVIEW

File: ci/scripts/build-bitstream-vivado.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+usage() {
```
Comment (mcy) [inline ci/scripts/build-bitstream-vivado.sh:8]: This seems sufficiently complicated that perhaps a python script is in order. (Certainly, if you're parsing flags...)

Comment (rswarbrick) [reply ci/scripts/build-bitstream-vivado.sh:8]: Well, we're not exactly parsing flags: there are just two strings that we're matching against a list. I *could* write a Python script, but I reckon that will be significantly more work. And you can't then run it with `bash -x ci/scripts/build-bitstream-vivado.sh`...

**Comment by rswarbrick** [comment]
Of course, as soon as I mark this as a draft, I finally get the script right... :-)

**Comment by rswarbrick** [comment]
@vogelpi: Could you take a look at this?

**Review by vogelpi — APPROVED** [review]
I saw this PR already a couple of days ago but it then failed CI. I definitely think this is a good idea. Thanks @rswarbrick for doing it, LGTM!

**Pull request merged**